spec/std/path_spec.cr

```diff
@@ -544,34 +552,49 @@ describe Path do
       it_expands_path("C:\\foo", "D:/C:\\foo", "C:\\foo", base: "D:/")
     end
 
+    describe "doesn't expand ~" do
+      path = Path["~"]
+      path.expand.should eq path
```

Comment:
@straight-shoota I don't think the review was correct.  The test makes sure `path.expand` without a home is **not** the same as with a home.  Also the spec fails.  

It's the last commit.  Should I undo it?
```
-      path.expand.should_not eq path.expand(home: "/foo")
+      path.expand.should eq path
```

Reply:
This spec is supposed to make sure that a `~` component will be left untouched by `#expand`.
In order to spec this correctly, we need to remove base expansion:
```cr
path.expand(base: "", expand_base: false).should eq path
```
And please also add a test for `Path["~/foo"]`